modules/solid_mechanics/src/bcs/InteractionIntegralBenchmarkBC.C

```diff
@@ -0,0 +1,92 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "InteractionIntegralBenchmarkBC.h"
+#include "Function.h"
+
+template<>
+InputParameters validParams<InteractionIntegralBenchmarkBC>()
+{
+  InputParameters params = validParams<PresetNodalBC>();
+  params.addRequiredParam<int>("component","The component of the displacement (0=x,1=y,2=z).");
+  params.addRequiredParam<UserObjectName>("crack_front_definition","The CrackFrontDefinition user object name");
+  params.addParam<unsigned int>("crack_front_node_index",0,"The index of the node on the crack front.");
+  params.addParam<Real>("poissons_ratio", "Poisson's ratio for the material.");
+  params.addParam<Real>("youngs_modulus", "Young's modulus of the material.");
+  params.addParam<Real>("KI",1.0,"Mode I stress intensity factor to apply.");
+  params.addParam<Real>("KII",1.0,"Mode II stress intensity factor to apply.");
+  params.addParam<Real>("KIII",1.0,"Mode III stress intensity factor to apply.");
+  return params;
+}
+
+InteractionIntegralBenchmarkBC::InteractionIntegralBenchmarkBC(const std::string & name, InputParameters parameters) :
+    PresetNodalBC(name, parameters),
+    _component(getParam<int>("component")),
+    _crack_front_definition(&getUserObject<CrackFrontDefinition>("crack_front_definition")),
+    _crack_front_node_index(getParam<unsigned int>("crack_front_node_index")),
+    _poissons_ratio(getParam<Real>("poissons_ratio")),
+    _youngs_modulus(getParam<Real>("youngs_modulus")),
+    _ki(getParam<Real>("KI")),
+    _kii(getParam<Real>("KII")),
+    _kiii(getParam<Real>("KIII"))
+{
+  if (_component < 0 || _component > 2)
+  {
+    std::stringstream errMsg;
+    errMsg << "Invalid component given for "
+           << name
+           << ": "
+           << _component
+           << "." << std::endl;
+
+    mooseError( errMsg.str() );
+  }
+}
```

Comment:
You can use a range checked param here: http://mooseframework.org/blog/range-checked-parameters/